server/webapp/WEB-INF/rails.new/webpack/single_page_apps/analytics.js

```diff
@@ -14,44 +14,51 @@
  * limitations under the License.
  */
 
-(function() {
-  "use strict";
+const $               = require('jquery');
+const m               = require('mithril');
+const Stream          = require('mithril/stream');
+const AnalyticsWidget = require('views/analytics/analytics_widget');
+const PluginInfos     = require('models/shared/plugin_infos');
+const VersionUpdater  = require('models/shared/version_updater');
+const PluginEndpoint  = require('rails-shared/plugin-endpoint');
+require('foundation-sites');
+require('helpers/server_health_messages_helper');
 
-  const m = require("mithril");
-  const $ = require("jquery");
-
-  require("foundation-sites");
-  require('helpers/server_health_messages_helper');
-
-  const PluginEndpoint           = require('rails-shared/plugin-endpoint');
-  const VersionUpdater           = require('models/shared/version_updater');
-  const Tab                      = require('models/analytics/tab');
-  const Tabs                     = require('models/analytics/tabs');
-  const AnalyticsDashboardHeader = require('views/analytics/header');
-  const DashboardTabs            = require('views/analytics/tabs');
-  const GlobalMetrics            = require('views/analytics/global_metrics');
-  const PipelineMetrics          = require('views/analytics/pipeline_metrics');
+$(() => {
+  $(document).foundation();
+  new VersionUpdater().update();
 
   PluginEndpoint.ensure("v1");
+  const analyticsElem = $('.analytics-container');
+  const pipelines     = JSON.parse(analyticsElem.attr('data-pipeline-list'));
 
-  document.addEventListener("DOMContentLoaded", () => {
-    const main = document.querySelector("[data-supported-dashboard-metrics]");
+  const onSuccess = (pluginInfos) => {
+    const plugins = pluginInfos.mapPluginInfos((p) => p.id());
+    const metrics = {};
+    pluginInfos.eachPluginInfo((p) => {
```

Comment:
Use `mapPluginInfo` instead?

Reply:
Can you provide an example, how to do it using map? I thought, we need a reduce, but the mixin does not provide reduce functionality.